src/decoder/vorbis.rs

```diff
@@ -44,12 +52,12 @@ impl<R> Source for VorbisDecoder<R> where R: Read + Seek {
 
     #[inline]
     fn get_channels(&self) -> u16 {
-        self.current_channels
+        self.stream_reader.ident_hdr.audio_channels as u16
```

Comment:
No, the number of channels is constant over the whole stream: https://www.xiph.org/vorbis/doc/Vorbis_I_spec.html#x1-120001.2.1